### DISCUSSION TITLE
Adding event filters via table

### DISCUSSION THREAD

**Discussion by irowebbn** [nasa/cFS#503] [Q&A] [answered]
Is it possible to add a default filter to events from apps like EVS or CFE_TIME via a table? If so, is there an example of that available?

**Comment by skliper** [comment, marked as answer]
If you are using Stored Command (https://github.com/nasa/SC) you could put the commands to add the filters in an RTS table that runs at startup.  You could even have multiple RTS's that have different filter sets that you could selectively run as needed.

**Reply by skliper** [reply]
IIRC by default RTS 0 runs for a "power on" reset and RTS 1 runs for "processor" reset (although it's configurable).  Any of them can be run by command.  There's example tables here, "just" change the noops to add event filter commands.
https://github.com/nasa/SC/blob/main/fsw/tables/sc_rts001.c

Note there is a macro getting added to cFE (nasa/cFE#2114) that supports an improvement in table definitions as demoed here (nasa/SC#35).
